libs/pubsub/query/query.go

```diff
@@ -452,42 +457,57 @@ func matchValue(value string, op Operator, operand reflect.Value) (bool, error)
 			return v == operandFloat64, nil
 		}
 
-	case reflect.Int64:
-		var v int64
+	case reflect.Pointer:
 
-		operandInt := operand.Interface().(int64)
-		filteredValue := numRegex.FindString(value)
+		switch operand.Interface().(type) {
+		case *big.Int:
+			filteredValue := numRegex.FindString(value)
+			operandVal := operand.Interface().(*big.Int)
+			v := new(big.Int)
+			if strings.ContainsAny(filteredValue, ".") {
+				// We do this just to check whether the string can be parsed as a float
+				_, err := strconv.ParseFloat(filteredValue, 64)
```

Comment:
What if we get a very big float (e.g., 600000000000000000000000000000000000000000000000000000001), shouldn't we still parse it as a BigFloat (to make sure we don't lose the last 1) ?